### PR TITLE
Use log functions of core framework on test/e2e/framework/kubelet

### PR DESCRIPTION
**Pull request by clarklee92**
Signed-off-by: clarklee92 <clarklee1992@hotmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This makes sub packages of e2e/framework/kubelet/* to use log functions
of core framework avoiding circular dependencies:
1. move GetOneTimeResourceUsageOnNode() from test/e2e/framework/kubelet/stats.go to getOneTimeResourceUsageOnNode() in test/e2e/framework/resource_usage_gatherer.go
2. copy GetKubeletPods() from test/e2e/framework/kubelet/kubelet_pods.go to getKubeletPods() in test/e2e/framework/util.go

**Which issue(s) this PR fixes**:

Ref: #81427

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/framework/resource_usage_gatherer.go [hunk 1]
```diff
@@ -192,6 +210,115 @@ func (w *resourceGatherWorker) singleProbe() {
...
+// cadvisor records cumulative cpu usage in nanoseconds, so we need to have two
+// stats points to compute the cpu usage over the interval. Assuming cadvisor
+// polls every second, we'd need to get N stats points for N-second interval.
+// Note that this is an approximation and may not be accurate, hence we also
+// write the actual interval used for calculation (based on the timestamps of
+// the stats points in ContainerResourceUsage.CPUInterval.
+//
+// containerNames is a function returning a collection of container names in which
+// user is interested in.
+func getOneTimeResourceUsageOnNode(
```
Comment (oomichi) [inline test/e2e/framework/resource_usage_gatherer.go]: confirmed GetOneTimeResourceUsageOnNode() is used in this package as https://cs.k8s.io/?q=GetOneTimeResourceUsageOnNode&i=nope&files=&repos=
So it is nice to move the function to here, could you write it on the PR message?

File: test/e2e/framework/kubelet/stats.go [hunk 2]
```diff
@@ -246,83 +246,6 @@ func removeUint64Ptr(ptr *uint64) uint64 {
 	return *ptr
 }
 
-// GetOneTimeResourceUsageOnNode queries the node's /stats/summary endpoint
```
Comment (oomichi) [inline test/e2e/framework/kubelet/stats.go]: We need to remove `removeUint64Ptr()` also for passing the following error:
```
Errors from staticcheck:
test/e2e/framework/kubelet/stats.go:242:6: func removeUint64Ptr is unused (U1000)
```
pull-kubernetes-verify job tends to be stable, that means we need to check the reason and log before running `/retest` as https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/84640/pull-kubernetes-verify/1190253504040538112

**Review by oomichi — COMMENTED** [review]
Thanks for doing this, I am happy to see this.
just few comments.

File: test/e2e/framework/kubelet/stats.go (hunk 2 above)
Comment (clarklee92) [reply test/e2e/framework/kubelet/stats.go]: Sorry about that, as `/retest` tagged on the phone and  I didn't read the log carefully.🤣

**Comment by oomichi** [comment]
Thanks for doing this, lgtm now

/lgtm
/approve

**Pull request merged**